tests/test_sequence.py

```diff
@@ -137,6 +138,325 @@ def test_magnetic_field():
+        channel_name="ising",
+        channel_id="rydberg_global",
+        l_pulses=pulses[:2],
+    )
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right phase_jump_time & clock_period.",
+    ):
+        seq.switch_device(MockDevice, True)
+
+    # Jump_phase_time check 2: No phase
+
+    seq = init_seq(
+        devices[2],
+        channel_name="ising",
+        channel_id="rydberg_global",
+        l_pulses=[pulses[0], pulses[2]],
+    )
+    with pytest.warns(
+        UserWarning,
+        match="The phase_jump_time of the matching channel"
+        + " on the the new device is different, take it into account"
+        + " for the upcoming pulses.",
+    ):
+        seq.switch_device(Chadoq2, True)
+
+    # Clock_period not match
+    seq = init_seq(
+        devices[0],
+        channel_name="ising",
+        channel_id="rydberg_global",
+        l_pulses=pulses[:2],
+    )
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel ising"
+        + " with the right phase_jump_time & clock_period.",
+    ):
+        seq.switch_device(devices[1], True)
+
+    seq = init_seq(
+        devices[2],
+        channel_name="digital",
+        channel_id="rmn_local1",
+        l_pulses=[],
+        initial_target=["q0"],
+    )
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel digital"
+        + " with the right mod_bandwidth.",
+    ):
+        seq.switch_device(devices[0], True)
+
+    with pytest.raises(
+        ValueError,
+        match="No channel match for channel digital"
+        + " with the right fixed_retarget_t.",
+    ):
+        seq.switch_device(devices[1], True)
+
+
+def test_switch_device_up(devices, pulses):
+
+    # Device checkout
+    seq = init_seq(Chadoq2, "ising", "rydberg_global", None)
+    assert seq.switch_device(Chadoq2)._device == Chadoq2
+
+    # Test non-strict mode
+    assert "ising" in seq.switch_device(devices[0]).declared_channels
+
+    # Strict: Jump_phase_time & CLock-period criteria
+    # Jump_phase_time check 1: phase not nill
+    seq1 = init_seq(
+        devices[2],
+        channel_name="ising",
+        channel_id="rydberg_global",
+        l_pulses=pulses[:2],
+    )
+    seq2 = init_seq(
+        devices[0],
+        channel_name="ising",
+        channel_id="rydberg_global",
+        l_pulses=pulses[:2],
+    )
+    new_seq = seq1.switch_device(devices[0], True)
+    s1 = sample(new_seq)
+    s2 = sample(seq1)
+    s3 = sample(seq2)
+    nested_s1 = s1.to_nested_dict()["Global"]["ground-rydberg"]
+    nested_s2 = s2.to_nested_dict()["Global"]["ground-rydberg"]
+    nested_s3 = s3.to_nested_dict()["Global"]["ground-rydberg"]
+
+    # Check if the samples are the same
+    for key in ["amp", "det", "phase"]:
+        np.testing.assert_array_equal(nested_s1[key], nested_s3[key])
+        np.testing.assert_array_equal(nested_s1[key], nested_s2[key])
```

Comment:
Continuing the example above, I think you'll need to add this:
```suggestion
        np.testing.assert_array_equal(nested_s1[key], nested_s3[key])
        if strict:
            np.testing.assert_array_equal(nested_s1[key], nested_s2[key])
```